### PR TITLE
feat(termwiz): add `From` termwiz style impls

### LINKED ISSUE #679: Implement `From` for `Style`
## Problem

There is no way to create a `Style` from crossterm, termion or termwiz style.

This is impossible to implement on the user's side because of the orphan rule. Or super cumbersome with a new-type pattern (not to mention underlying `Color` and attributes have the exact same problem).

It is also impossible to use raw ansi code when creating `Text`, `Line` or `Span` as they will render 
image
Code reference:
image

## Solution

Add a `From` impl for `Style`, `Color` and `Modifier`.
These should be under their corresponding terminal feature.

~~We could also authorize raw ansi codes when creating `Text`, `Line`, `Span` and others (if this is possible, don't know the underlying limitations here).~~

EDIT for future readers: [ansi-to-tui](https://crates.io/crates/ansi-to-tui) does exactly that.

## Additional context

I wanted to use https://github.com/sharkdp/lscolors to retrieve `LS_COLORS` and print a list of files.

This could somewhat relate to #493. 

# Tasks

- [x] `crossterm`
- [x] `termion`
- [ ] `termwiz`

### PR DESCRIPTION
**Pull request by Valentin271**
Closes #679

Important note: this also fixes a wrong mapping between ratatui's gray and termwiz's grey. `ratatui::Color::Gray` now maps to `termwiz::color::AnsiColor::Silver`

### CODE REVIEW

File: src/backend/termwiz.rs [hunk 1]
```diff
@@ -249,12 +249,73 @@ impl Backend for TermwizBackend {
...
+}
+
 impl From<Color> for ColorAttribute {
     fn from(color: Color) -> ColorAttribute {
         match color {
             Color::Reset => ColorAttribute::Default,
             Color::Black => AnsiColor::Black.into(),
-            Color::Gray | Color::DarkGray => AnsiColor::Grey.into(),
+            Color::DarkGray => AnsiColor::Grey.into(),
+            Color::Gray => AnsiColor::Silver.into(),
```
Comment (Valentin271) [inline src/backend/termwiz.rs:318]: | ANSI | ratatui | termwiz |
|--------|--------|--------|
| white (37) | Gray <sup>1</sup> | Silver <sup>2</sup> |
| bright black (90) | DarkGray <sup>3</sup> | Grey | 

1: As stated in our doc "_this is sometimes called `silver` or `white` but we use `white` for bright white_"
2: As stated in [termwiz doc](https://docs.rs/termwiz/latest/termwiz/color/enum.AnsiColor.html#variant.Silver), silver is "dark white". "dark" in termwiz actually means normal in ANSI

3: As stated in our doc "_this is sometimes called `light black` or `bright black` but we use `dark gray`_"
4: According to [termwiz doc](https://docs.rs/termwiz/latest/termwiz/color/enum.AnsiColor.html#variant.Grey), grey is "bright black"

Comment (joshka) [reply src/backend/termwiz.rs:318]: This would probably be a breaking change, or worth noting in the changelog specifically.

**Review by joshka — APPROVED** [review]
LGTM, though when merging this make sure to document the color change for the changelog.

File: src/backend/termwiz.rs (hunk 1 above)
Comment (Valentin271) [reply src/backend/termwiz.rs:318]: It *is* a breaking changing, but unlikely to hit anyone in a breaking way. Definitely worth mentionning in the changelog though 👍

**Pull request merged**